iac-extensions/spring-config/src/main/antlr/PropertiesLexer.g4

```diff
@@ -7,13 +7,19 @@ public PropertiesLexer(CharStream input, boolean crLexerCostructor) {
 }
 }
 
-COMMENT         : {this.getCharPositionInLine() == 0}? [!#] -> pushMode(VALUE_MODE);
+COMMENT         : [!#] -> pushMode(COMMENT_MODE);
 LEADING_SPACING : {this.getCharPositionInLine() == 0}? [ \t\f\r\n\u2028\u2029]+ -> channel(HIDDEN);
 NEWLINE         : [\r\n\u2028\u2029]+;
 DELIMITER       : [ ]* [:=\t\f ] [ ]* -> pushMode(VALUE_MODE);
 SLASH           : '\\' -> more, pushMode(INSIDE);
 CHARACTER       : ~ [:=\r\n\u2028\u2029];
 
+mode COMMENT_MODE;
+
+COMMENT_NEW_LINE  : [\r\n\u2028\u2029]+      -> type(NEWLINE), popMode;
+COMMENT_DELIMITER : [ ]* [:=\t\f ] [ ]*      -> type(DELIMITER), popMode;
```

Comment:
Why do we need to `popMode` after delimiters in comments? Shouldn't it happen only after new lines?